### DISCUSSION TITLE
Why Does Fooocus NOT have 16:9 Ratio?

### DISCUSSION THREAD

**Discussion by n2lee** [lllyasviel/Fooocus#3131] [Q&A] [answered]
First let me say, I have edited the modules file to support 1920x1080 but every time there is an update (which I applaud) it requires editing the modules file.

But with 16:9 being the most common aspect ratio on the planet, why is this not an option?

Request:
1. Add 1920x1080
2. Give the user the ability to select their default so each time the system starts their preferences are already loaded?

**Comment by monstamash77** [comment, marked as answer]
1) SDXL models are trained on smaller images, 1024*1024 and variations of this. For 16:9 use 1365 x 768 then upscale from there. See https://www.reddit.com/r/StableDiffusion/comments/15c3rf6/sdxl_resolution_cheat_sheet/ for more details. Nothing stopping you from generating at 1920x1080 but results may be strange
2) You can do this already by editing the config.txt file found in the root of Fooocus, or by creating your own presets from the presets folder

**Comment by xhoxye** [comment]
Because there is no such aspect ratio in the dataset when sdxl is trained
https://github.com/Stability-AI/generative-models/blob/477d8b9a7730d9b2e92b326a770c0420d00308c9/scripts/demo/sampling.py#L7
You can use this aspect ratio, but the result is likely to collapse
You can avoid updating by editing config.txt profile

**Comment by Alexdnk** [comment]
@n2lee  add these lines to your separate custom preset:
```
	"available_aspect_ratios": [
        "704*1408",
        "704*1344",
        "768*1344",
        "768*1280",
        "832*1216",
        "832*1152",
        "896*1152",
        "896*1088",
        "960*1088",
        "960*1024",
        "1024*1024",
        "1024*960",
        "1088*960",
        "1088*896",
        "1152*896",
        "1152*832",
        "1216*832",
        "1280*768",
        "1344*768",
        "1344*704",
        "1408*704",
        "1472*704",
        "1536*640",
        "1600*640",
        "1664*576",
        "1728*576",
        "1920*1080",
        "1080*1920"
    ],
```

@monstamash77 @xhoxye  for example, with [this](https://huggingface.co/dataautogpt3/TempestV0.1) model it is recommended:

> supported sizes: | 2048x1024 | 1920x1088 |

**Comment by mashb1t** [comment]
Also see https://github.com/lllyasviel/Fooocus/pull/1617#issuecomment-1870469961 and below where this question has already been answered.

**Comment by n2lee** [comment]
Thanks for the great answers. I admit I do not understand this issue of training models. I thought these gen ai systems worked on the pixel level not the image level. When I look at the aspect ratios that are offered, it is obvious that most of these are not normal aspect ratios. Following this logic then the models would all have to be trained on these odd aspect ratios.  Oh Well no biggie.

I guess I haven't upgraded in awhile. When was the structure changed and the settings moved out of the config.py file ?

QUESTION:
When adding this new info to the custom preset file will this prevent this file from getting over ridden during the next update?

Thanks again for the great info.

**Reply by mashb1t** [reply]
custom presets are never touched/modified/etc., so they're safe to use. The config is also only changed when there are issues with the values in it.

**Comment by DavidDragonsage** [comment]
@n2lee I will indulge in a little self-advertising here and point you to the Pure Fooocus aspect ratio guide at Facebook: 
https://www.facebook.com/groups/fooocus/learning_content/?filter=800997985311610
Facebook linking is not reliable, but the guide is called "Aspect Ratios & Artistic Expression" if the link doesn't quite get you there.

I have found that 9:16 and 16:9 are typically quite reliable, provided you do not exceed the SDXL maximum resolution of 1024x1024 pixels, i.e. 1,048,576 pixels (the area bounded by 1024x1024). But within that area there are 40 "legal" aspect ratio variations, of which Fooocus only includes 26. Every legal variation is a multiple of 64 pixels, in both the horizontal and vertical dimensions.

Anyway, I think I had better just leave it there, otherwise this reply is going to grow to an unmanageable size if I keep quoting from the guide! 😏
